include/crow/json.h

```diff
@@ -212,17 +212,44 @@ namespace crow
 
             inline bool operator==(const r_string& l, const r_string& r)
             {
-                return std::equal(l.begin(), l.end(), r.begin(), r.end());
+                if (l.size() != r.size())
+                    return false;
+
+                for (size_t i = 0; i < l.size(); i++)
+                {
+                    if (*(l.begin() + i) != *(r.begin() + i))
+                        return false;
+                }
+
+                return true;
```

Comment:
would a template utility function that uses `std::equals` for newer C++ and this function for C++11 work?

Reply:
I guess it would but if the algorithm has to be there anyway then why not use it.

Reply:
is there a way to have this algorithm in a way where it's not repeated for all 3 `==` operators?

Reply:
I could see a private templated function that is being called by these operators.